selfservice/hook/web_hook.go

```diff
@@ -499,3 +504,23 @@ func isTimeoutError(err error) bool {
 	var te interface{ Timeout() bool }
 	return errors.As(err, &te) && te.Timeout() || errors.Is(err, context.DeadlineExceeded)
 }
+
+func instrumentHTTPClientForEvents(ctx context.Context, httpClient *retryablehttp.Client) {
+	var (
+		attempt   = 0
+		requestID uuid.UUID
+		reqBody   []byte
+	)
+	httpClient.RequestLogHook = func(_ retryablehttp.Logger, req *http.Request, retryNumber int) {
+		attempt = retryNumber + 1
+		requestID = uuid.Must(uuid.NewV4())
```

Comment:
`Ory-Network-Request-Id` identifies the incoming request though, doesn't it? The point of this header is to allow the webhook receiver to differentiate between different attempts (retries).

Reply:
Ah I see, but shouldn't we also add a way to correlate attempts, i.e. the `Ory-Network-Request-Id`? This would allow us to better correlate all attempts into one.

Reply:
This would be useful for all events and should go into https://github.com/ory/x/blob/master/otelx/semconv/events.go.

Out of scope for this PR.